ttg/ttg/terminal.h

```diff
@@ -11,8 +11,80 @@
 #include "ttg/util/trace.h"
 #include "ttg/util/demangle.h"
 #include "ttg/world.h"
+#include "boost/callable_traits.hpp"
 
 namespace ttg {
+  namespace detail {
+
+    /* Wraps any key,value data structure.
+     * Elements of the data structure can be accessed using get method, which calls the at method of the Container.
+     * keyT - taskID
+     * valueT - Value type of the Container
+    */
+    template<typename keyT, typename valueT>
+    struct ContainerWrapper {
+      std::function<valueT (keyT const& key)> get = nullptr;
+      std::function<size_t (keyT const& key)> owner = nullptr;
+
+      ContainerWrapper() = default;
+
+      template<typename T, typename mapperT, typename keymapT, std::enable_if_t<!std::is_same<std::decay_t<T>,
+                                                          ContainerWrapper>{}, bool> = true>
+        //Store a pointer to the user's container in std::any, no copies
+        ContainerWrapper(T &t, mapperT &&mapper,
+                         keymapT &&keymap) : get([&t,&mapper](keyT const &key) {
+                                                  if constexpr (!std::is_class_v<T> && std::is_function_v<T>) {
```

Comment:
Do you mean `std::is_invocable_v` instead of `std::is_function_v`? https://en.cppreference.com/w/cpp/types/is_function

Reply:
I think we should capture `t` and `mapper` by value. Otherwise bad things happen if they go out of scope (e.g., at the end of a `make_my_task` function)